diff -ur monero-0.18.1.0-a/CMakeLists.txt monero-0.18.1.0-b/CMakeLists.txt
--- monero-0.18.1.0-a/CMakeLists.txt	2022-08-10 03:34:23.000000000 +0000
+++ monero-0.18.1.0-b/CMakeLists.txt	2022-08-15 08:29:38.412362713 +0000
@@ -458,7 +458,7 @@
 # elseif(CMAKE_SYSTEM_NAME MATCHES ".*BSDI.*")
 #   set(BSDI TRUE)
 
-include_directories(external/rapidjson/include external/easylogging++ src contrib/epee/include external external/supercop/include)
+include_directories(external/rapidjson/include external/easylogging++ src contrib/epee/include external)
 
 if(APPLE)
   cmake_policy(SET CMP0042 NEW)
@@ -1201,7 +1201,6 @@
   set(ZMQ_LIB "${ZMQ_LIB};${SODIUM_LIBRARY}")
 endif()
 
-include(external/supercop/functions.cmake) # place after setting flags and before src directory inclusion
 add_subdirectory(contrib)
 add_subdirectory(src)
 
diff -ur monero-0.18.1.0-a/external/CMakeLists.txt monero-0.18.1.0-b/external/CMakeLists.txt
--- monero-0.18.1.0-a/external/CMakeLists.txt	2022-08-10 03:34:23.000000000 +0000
+++ monero-0.18.1.0-b/external/CMakeLists.txt	2022-08-15 08:30:20.328571977 +0000
@@ -37,21 +37,7 @@
 
 find_package(Miniupnpc REQUIRED)
 
-message(STATUS "Using in-tree miniupnpc")
-set(UPNPC_NO_INSTALL TRUE CACHE BOOL "Disable miniupnp installation" FORCE)
-add_subdirectory(miniupnp/miniupnpc)
-set_property(TARGET libminiupnpc-static PROPERTY FOLDER "external")
-set_property(TARGET libminiupnpc-static PROPERTY POSITION_INDEPENDENT_CODE ON)
-if(MSVC)
-  set_property(TARGET libminiupnpc-static APPEND_STRING PROPERTY COMPILE_FLAGS " -wd4244 -wd4267")
-elseif(NOT MSVC)
-  set_property(TARGET libminiupnpc-static APPEND_STRING PROPERTY COMPILE_FLAGS " -Wno-undef -Wno-unused-result -Wno-unused-value")
-endif()
-if(CMAKE_SYSTEM_NAME MATCHES "NetBSD")
-	set_property(TARGET libminiupnpc-static APPEND_STRING PROPERTY COMPILE_FLAGS " -D_NETBSD_SOURCE")
-endif()
-
-set(UPNP_LIBRARIES "libminiupnpc-static" PARENT_SCOPE)
+set(UPNP_LIBRARIES "miniupnpc" PARENT_SCOPE)
 
 find_package(Unbound)
 
@@ -69,4 +55,3 @@
 add_subdirectory(db_drivers)
 add_subdirectory(easylogging++)
 add_subdirectory(qrcodegen)
-add_subdirectory(randomx EXCLUDE_FROM_ALL)
diff -ur monero-0.18.1.0-a/src/crypto/CMakeLists.txt monero-0.18.1.0-b/src/crypto/CMakeLists.txt
--- monero-0.18.1.0-a/src/crypto/CMakeLists.txt	2022-08-10 03:34:23.000000000 +0000
+++ monero-0.18.1.0-b/src/crypto/CMakeLists.txt	2022-08-15 08:31:12.034830124 +0000
@@ -55,6 +55,7 @@
 endif()
 
 include_directories(${RANDOMX_INCLUDE})
+include_directories(SYSTEM /usr/include/monero)
 
 set(crypto_headers)
 
@@ -97,6 +98,3 @@
 # cheat because cmake and ccache hate each other
 set_property(SOURCE CryptonightR_template.S PROPERTY LANGUAGE C)
 set_property(SOURCE CryptonightR_template.S PROPERTY XCODE_EXPLICIT_FILE_TYPE sourcecode.asm)
-
-# Must be done last, because it references libraries in this directory
-add_subdirectory(wallet)
diff -ur monero-0.18.1.0-a/src/crypto/wallet/crypto.h monero-0.18.1.0-b/src/crypto/wallet/crypto.h
--- monero-0.18.1.0-a/src/crypto/wallet/crypto.h	2022-08-10 03:34:23.000000000 +0000
+++ monero-0.18.1.0-b/src/crypto/wallet/crypto.h	2022-08-15 08:31:30.087920252 +0000
@@ -30,7 +30,6 @@
 #pragma once
 
 #include <cstddef>
-#include "crypto/wallet/ops.h"
 
 namespace crypto {
   namespace wallet {
diff -ur monero-0.18.1.0-a/src/device/CMakeLists.txt monero-0.18.1.0-b/src/device/CMakeLists.txt
--- monero-0.18.1.0-a/src/device/CMakeLists.txt	2022-08-10 03:34:23.000000000 +0000
+++ monero-0.18.1.0-b/src/device/CMakeLists.txt	2022-08-15 08:31:49.795018639 +0000
@@ -73,7 +73,7 @@
     cncrypto
     cryptonote_format_utils_basic
     ringct_basic
-    wallet-crypto
+    monero-crypto
     ${OPENSSL_CRYPTO_LIBRARIES}
     ${Boost_SERIALIZATION_LIBRARY}
   PRIVATE
diff -ur monero-0.18.1.0-a/src/p2p/net_node.inl monero-0.18.1.0-b/src/p2p/net_node.inl
--- monero-0.18.1.0-a/src/p2p/net_node.inl	2022-08-10 03:34:23.000000000 +0000
+++ monero-0.18.1.0-b/src/p2p/net_node.inl	2022-08-15 08:32:15.702147978 +0000
@@ -60,9 +60,9 @@
 #include "cryptonote_core/cryptonote_core.h"
 #include "net/parse.h"
 
-#include <miniupnp/miniupnpc/miniupnpc.h>
-#include <miniupnp/miniupnpc/upnpcommands.h>
-#include <miniupnp/miniupnpc/upnperrors.h>
+#include <miniupnpc/miniupnpc.h>
+#include <miniupnpc/upnpcommands.h>
+#include <miniupnpc/upnperrors.h>
 
 #undef MONERO_DEFAULT_LOG_CATEGORY
 #define MONERO_DEFAULT_LOG_CATEGORY "net.p2p"
