diff --git a/Build.PL b/Build.PL
index 8f882fc..f99e978 100644
--- a/Build.PL
+++ b/Build.PL
@@ -100,15 +100,6 @@ EOF
     my @cpanm_args = ();
     push @cpanm_args, "--sudo" if $sudo;
 
-    # install local::lib without --local-lib otherwise it's not usable afterwards
-    if (!eval "use local::lib qw(local-lib); 1") {
-        my $res = system $cpanm, @cpanm_args, 'local::lib';
-        warn "Warning: local::lib is required. You might need to run the `cpanm --sudo local::lib` command in order to install it.\n"
-            if $res != 0;
-    }
-    
-    push @cpanm_args, ('--local-lib', 'local-lib') if ! $nolocal;
-
     # make sure our cpanm is updated (old ones don't support the ~ syntax)
     system $cpanm, @cpanm_args, 'App::cpanminus';
     
diff --git a/lib/Slic3r.pm b/lib/Slic3r.pm
index 0c6c81b..625187b 100644
--- a/lib/Slic3r.pm
+++ b/lib/Slic3r.pm
@@ -41,7 +41,7 @@ warn "Running Slic3r under Perl 5.16 is neither supported nor recommended\n"
 use FindBin;
 
 # Let the XS module know where the GUI resources reside.
-set_resources_dir(decode_path($FindBin::Bin) . (($^O eq 'darwin') ? '/../Resources' : '/resources'));
+set_resources_dir(decode_path($FindBin::RealBin) . (($^O eq 'darwin') ? '/../Resources' : '/resources'));
 set_var_dir(resources_dir() . "/icons");
 set_local_dir(resources_dir() . "/localization/");
 
diff --git a/slic3r.pl b/slic3r.pl
index d9bed0a..f4c29df 100755
--- a/slic3r.pl
+++ b/slic3r.pl
@@ -6,7 +6,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/lib";
-    use local::lib '--no-create', "$FindBin::Bin/local-lib";
 }
 
 use File::Basename qw(basename);
diff --git a/t/angles.t b/t/angles.t
index 9dc690d..363d154 100644
--- a/t/angles.t
+++ b/t/angles.t
@@ -6,8 +6,7 @@ plan tests => 34;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use Slic3r;
@@ -90,4 +89,4 @@ sub line_orientation {
 sub line_direction {
     my ($l) = @_;
     return Slic3r::Line->new(@$l)->direction;
-}
\ No newline at end of file
+}
diff --git a/t/bridges.t b/t/bridges.t
index ca55862..7ff7691 100644
--- a/t/bridges.t
+++ b/t/bridges.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first sum);
diff --git a/t/clean_polylines.t b/t/clean_polylines.t
index 50c6f5b..d2f1cac 100644
--- a/t/clean_polylines.t
+++ b/t/clean_polylines.t
@@ -6,8 +6,7 @@ plan tests => 6;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use Slic3r;
diff --git a/t/clipper.t b/t/clipper.t
index 3c98381..e5f5d6a 100644
--- a/t/clipper.t
+++ b/t/clipper.t
@@ -6,8 +6,7 @@ plan tests => 6;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(sum);
diff --git a/t/collinear.t b/t/collinear.t
index b28a360..b715b2d 100644
--- a/t/collinear.t
+++ b/t/collinear.t
@@ -6,8 +6,7 @@ plan tests => 11;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use Slic3r;
diff --git a/t/combineinfill.t b/t/combineinfill.t
index 5402a84..02edf85 100644
--- a/t/combineinfill.t
+++ b/t/combineinfill.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/config.t b/t/config.t
index f4a1867..6228b6a 100644
--- a/t/config.t
+++ b/t/config.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use Slic3r;
diff --git a/t/cooling.t b/t/cooling.t
index 2f444cf..a2c83c4 100644
--- a/t/cooling.t
+++ b/t/cooling.t
@@ -6,8 +6,7 @@ plan tests => 14;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(none all);
diff --git a/t/custom_gcode.t b/t/custom_gcode.t
index 4c1c1b1..7decbf4 100644
--- a/t/custom_gcode.t
+++ b/t/custom_gcode.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/dynamic.t b/t/dynamic.t
index 5d4d3ce..b0eae7d 100644
--- a/t/dynamic.t
+++ b/t/dynamic.t
@@ -7,8 +7,7 @@ plan tests => 20;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/fill.t b/t/fill.t
index dd9eee4..50d1eaa 100644
--- a/t/fill.t
+++ b/t/fill.t
@@ -8,8 +8,7 @@ plan tests => 19;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first sum);
diff --git a/t/flow.t b/t/flow.t
index 2fa0d8f..ab146fd 100644
--- a/t/flow.t
+++ b/t/flow.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first sum);
diff --git a/t/gaps.t b/t/gaps.t
index f0c75c3..b122558 100644
--- a/t/gaps.t
+++ b/t/gaps.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/gcode.t b/t/gcode.t
index 126c621..5c5ee57 100644
--- a/t/gcode.t
+++ b/t/gcode.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/geometry.t b/t/geometry.t
index 0f37c0a..9fda622 100644
--- a/t/geometry.t
+++ b/t/geometry.t
@@ -6,8 +6,7 @@ plan tests => 42;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use Slic3r;
diff --git a/t/layers.t b/t/layers.t
index a9f7dfe..3ead931 100644
--- a/t/layers.t
+++ b/t/layers.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/loops.t b/t/loops.t
index e662469..eb956f6 100644
--- a/t/loops.t
+++ b/t/loops.t
@@ -7,8 +7,7 @@ plan tests => 4;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use Slic3r;
diff --git a/t/multi.t b/t/multi.t
index 75ce0c2..db8525a 100644
--- a/t/multi.t
+++ b/t/multi.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/perimeters.t b/t/perimeters.t
index ee33261..c5051ca 100644
--- a/t/perimeters.t
+++ b/t/perimeters.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use Slic3r::ExtrusionLoop ':roles';
diff --git a/t/polyclip.t b/t/polyclip.t
index 0808c7b..083af19 100644
--- a/t/polyclip.t
+++ b/t/polyclip.t
@@ -6,8 +6,7 @@ plan tests => 18;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use Slic3r;
diff --git a/t/print.t b/t/print.t
index 6939d5f..376ae97 100644
--- a/t/print.t
+++ b/t/print.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/retraction.t b/t/retraction.t
index 6e6a130..c0e3a74 100644
--- a/t/retraction.t
+++ b/t/retraction.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(any);
diff --git a/t/shells.t b/t/shells.t
index eac95aa..737f210 100644
--- a/t/shells.t
+++ b/t/shells.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first sum);
diff --git a/t/skirt_brim.t b/t/skirt_brim.t
index 225b0a9..5c7ece7 100644
--- a/t/skirt_brim.t
+++ b/t/skirt_brim.t
@@ -4,8 +4,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/slice.t b/t/slice.t
index 2f193aa..393709f 100644
--- a/t/slice.t
+++ b/t/slice.t
@@ -7,8 +7,7 @@ plan tests => 16;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 # temporarily disable compilation errors due to constant not being exported anymore
diff --git a/t/support.t b/t/support.t
index a0cac14..c5ea34a 100644
--- a/t/support.t
+++ b/t/support.t
@@ -7,8 +7,7 @@ plan tests => 27;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use List::Util qw(first);
diff --git a/t/svg.t b/t/svg.t
index 47d24ce..e610ab8 100644
--- a/t/svg.t
+++ b/t/svg.t
@@ -5,7 +5,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use Slic3r;
diff --git a/t/thin.t b/t/thin.t
index 9147236..56c4fe9 100644
--- a/t/thin.t
+++ b/t/thin.t
@@ -5,7 +5,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use Slic3r;
diff --git a/t/threads.t b/t/threads.t
index 7fede33..20b5bc4 100644
--- a/t/threads.t
+++ b/t/threads.t
@@ -5,7 +5,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use List::Util qw(first);
diff --git a/utils/amf-to-stl.pl b/utils/amf-to-stl.pl
index 802fd9a..2648ba2 100755
--- a/utils/amf-to-stl.pl
+++ b/utils/amf-to-stl.pl
@@ -7,7 +7,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use File::Basename qw(basename);
diff --git a/utils/dump-stl.pl b/utils/dump-stl.pl
index eee3c73..0f459ff 100644
--- a/utils/dump-stl.pl
+++ b/utils/dump-stl.pl
@@ -8,7 +8,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use Slic3r;
diff --git a/utils/pdf-slices.pl b/utils/pdf-slices.pl
index ca61da0..670a171 100755
--- a/utils/pdf-slices.pl
+++ b/utils/pdf-slices.pl
@@ -7,7 +7,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use Getopt::Long qw(:config no_auto_abbrev);
diff --git a/utils/send-gcode.pl b/utils/send-gcode.pl
index 0b803ba..989b178 100644
--- a/utils/send-gcode.pl
+++ b/utils/send-gcode.pl
@@ -6,7 +6,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use Slic3r;
diff --git a/utils/split_stl.pl b/utils/split_stl.pl
index 56217de..fffc576 100755
--- a/utils/split_stl.pl
+++ b/utils/split_stl.pl
@@ -7,7 +7,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use File::Basename qw(basename);
diff --git a/utils/stl-to-amf.pl b/utils/stl-to-amf.pl
index bb88b21..f0d537b 100755
--- a/utils/stl-to-amf.pl
+++ b/utils/stl-to-amf.pl
@@ -7,7 +7,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use File::Basename qw(basename);
diff --git a/utils/view-mesh.pl b/utils/view-mesh.pl
index 03153ab..66f1f3e 100644
--- a/utils/view-mesh.pl
+++ b/utils/view-mesh.pl
@@ -7,7 +7,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use Getopt::Long qw(:config no_auto_abbrev);
diff --git a/utils/view-toolpaths.pl b/utils/view-toolpaths.pl
index e064885..2a3dabc 100755
--- a/utils/view-toolpaths.pl
+++ b/utils/view-toolpaths.pl
@@ -6,8 +6,7 @@ use warnings;
 
 BEGIN {
     use FindBin;
-    use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
+    use lib "$FindBin::Bin/../lib";    
 }
 
 use Getopt::Long qw(:config no_auto_abbrev);
diff --git a/utils/wireframe.pl b/utils/wireframe.pl
index 5399f31..53ac69a 100644
--- a/utils/wireframe.pl
+++ b/utils/wireframe.pl
@@ -8,7 +8,6 @@ use warnings;
 BEGIN {
     use FindBin;
     use lib "$FindBin::Bin/../lib";
-    use local::lib "$FindBin::Bin/../local-lib";
 }
 
 use Getopt::Long qw(:config no_auto_abbrev);
